include/nbl/system/IFileArchive.h

```diff
@@ -6,19 +6,46 @@
 #ifndef __NBL_I_FILE_ARCHIVE_H_INCLUDED__
 #define __NBL_I_FILE_ARCHIVE_H_INCLUDED__
 
+
 #include "nbl/system/IFile.h"
 #include "nbl/system/CFileView.h"
-
+#include "nbl/system/IFileViewAllocator.h"
+#include "nbl/system/CFileViewVirtualAllocatorWin32.h"
 namespace nbl
 {
 
 namespace system
 {
 
+template<typename T>
+class CInnerArchiveFile : public CFileView<T>
+{
+	std::atomic_flag* alive;
+public:
+	CInnerArchiveFile(CFileView<T>* arch, std::atomic_flag* _flag) : CFileView<T>(std::move(*arch)), alive(_flag)
+	{
+		alive->test_and_set();
+		alive->notify_one();
+	}
+	~CInnerArchiveFile()
+	{
+		alive->clear();
+		alive->notify_one();
+	}
+};
```

Comment:
where's the `delete` overload?

Otherwise `drop()` will call a regular `delete` and call FREE on the memory 🤦